crawler/cv/pipelines/raw_data.py

```diff
@@ -0,0 +1,45 @@
+import re
+from cv.models.raw_data import RawData
+
+
+class RawDataPipeline(object):
+    def process_item(self, item, spider):
+        item["url"] = self.remove_slash(item["url"])
+        RawData.create(**item)
+        self.parse_entry(item)
+
+    def parse_entry(self, entry):
+        if entry["html"] is None:
+            return
+        # TODO how to remove url with a file extension?
+        # a possible way is using response.headers["Content-type"]
+        links = re.findall('(https?://'+entry["domain"]+'[^"#]*)', entry["html"])
+
+        # remove last / in the url
+        links = [self.remove_slash(x) for x in links]
+
+        # duplicate
+        links = list(set(links))
+
+        # blacklist regex
+        black_list = [
```

Comment:
This will be replace by some algorithm like [editdistance](https://pypi.python.org/pypi/editdistance)